codegen/module.go

```diff
@@ -303,6 +296,86 @@ func (system *ModuleSystem) ResolveModules(
 	return resolvedModules, nil
 }
 
+func (system *ModuleSystem) resolveMultiModules(
+	packageRoot string,
+	baseDirectory string,
+	targetGenDir string,
+	classDir string, // full path
+	className string,
+	class *ModuleClass,
+) ([]*ModuleInstance, error) {
+	classInstances := []*ModuleInstance{}
+
+	if classDir == "" {
+		return classInstances, nil
+	}
+
+	files, err := ioutil.ReadDir(classDir)
+
+	if err != nil {
+		// TODO: We should accumulate errors and list them all here
+		// Expected $path to be a class directory
+		return nil, errors.Wrapf(
+			err,
+			"Error reading module instance directory %q",
+			classDir,
+		)
+	}
+
+	relClassDir, err := filepath.Rel(baseDirectory, classDir)
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"Error relative class directory for %q",
+			className,
+		)
+	}
+
+outer:
+	for _, file := range files {
+		for _, dir := range class.SubDirs {
```

Comment:
This will probably be cleaner if you have an IsSubDir function so you can avoid the nested for loop and the labelled break.